encode.go

```diff
@@ -36,9 +36,13 @@ func (p *Part) Encode(writer io.Writer) error {
 	cte := p.setupMIMEHeaders()
 	// Encode this part.
 	b := bufio.NewWriter(writer)
-	p.encodeHeader(b)
+	if err := p.encodeHeader(b); err != nil {
+		return err
+	}
 	if len(p.Content) > 0 {
-		b.Write(crnl)
+		if _, err := b.Write(crnl); err != nil {
```

Comment:
for all of the b.Writes in this func, it doesn't matter if they fail, as it will cause the b.Flush at the end to fail.  can we do a `_, _ :=` instead to remove the iferr noise?  Or disable linting in this part?  It makes it really hard to read.

Reply:
I'm not sure that we can ignore these errors because `bufio.NewWriter` can be changed in the future and `b.Flush` will not check previous errors. What do you think?

Reply:
That's not correct, because Flush() does not exist on other writers, so it will be obvious if the writer type were to be changed.

Reply:
Will resolve this for now, I'd like to submit this PR so that other PRs don't have to deal with the linter errors.